library/data/registry-entry-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/api/RegistryEntryAttribute.java

```diff
@@ -0,0 +1,289 @@
+	 * @param <R> registry entry type
+	 * @param <V> value type
+	 * @return a new attribute
+	 */
+	static <R, V extends DispatchedType> RegistryEntryAttribute<R, V> createDispatched(Registry<R> registry, Identifier id,
+																					   Function<Identifier, Codec<? extends V>> codecGetter) {
+		return createDispatched(registry, id, codecGetter, null);
+	}
+
+	/**
+	 * Creates a new {@code boolean} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Boolean> createBool(Registry<R> registry, Identifier id, boolean defaultValue) {
+		return create(registry, id, Codec.BOOL, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code int} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Integer> createInt(Registry<R> registry, Identifier id, int defaultValue) {
+		return create(registry, id, Codec.INT, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code long} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Long> createLong(Registry<R> registry, Identifier id, long defaultValue) {
+		return create(registry, id, Codec.LONG, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code float} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Float> createFloat(Registry<R> registry, Identifier id, float defaultValue) {
+		return create(registry, id, Codec.FLOAT, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code double} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Double> createDouble(Registry<R> registry, Identifier id, double defaultValue) {
+		return create(registry, id, Codec.DOUBLE, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code String} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, String> createString(Registry<R> registry, Identifier id, String defaultValue) {
+		return create(registry, id, Codec.STRING, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code boolean} property.
```

Comment:
```suggestion
	 * Creates a new {@code boolean} property with no default value.
```

We should probably note the implicit contract this function declares and all others like it.